### PR TITLE
HDDS-4756. Add lock for activate/deactivate in SCMPipelineManager

### PR DESCRIPTION
**Pull request by Xushaohong**
## What changes were proposed in this pull request? 
Problem: In PipelineManager, the methods of activating and deactivating pipeline methods don't have the lock that originally supposed to have. Thus the result is that activate and activate operation could be requested concurrently and may lead to reading the stale version of the pipeline state. 

Solution: Add the write lock each time before calling the statemanager.

## What is the link to the Apache JIRA 
https://issues.apache.org/jira/browse/HDDS-4756 

## How was this patch tested?
The locked version is tested based on the tps upper bound test. The result shows that the number of ratis calls per second is declined rapidly.

### CODE REVIEW

**Comment by amaliujia** [comment]
cc @GlenGeng

**Comment by amaliujia** [comment]
@Xushaohong which is the final version for HDDS-4756? This PR or https://github.com/apache/ozone/pull/1862?

**Comment by Xushaohong** [comment]
> @Xushaohong which is the final version for [HDDS-4756](https://issues.apache.org/jira/browse/HDDS-4756)? This PR or #1862?

Thanks @amaliujia , I submited two PR, one is for master , and the other is for HDDS-2823.

**Review by GlenGeng — COMMENTED** [review]
LGTM.

@nandakumar131  @bshashikant  Please take a look! This is a bug found during test of SCM HA, which exists in master as well.

Please trigger an all green CI.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/SCMPipelineManager.java [hunk 1]
```diff
@@ -557,10 +557,15 @@ public void triggerPipelineCreation() {
...
-            getPipeline(pipelineID).getPipelineState();
-    stateManager.activatePipeline(pipelineID);
-    updatePipelineStateInDb(pipelineID, state);
+    lock.writeLock().lock();
+    try {
+      Pipeline.PipelineState state = stateManager.
+              getPipeline(pipelineID).getPipelineState();
+      stateManager.activatePipeline(pipelineID);
+      updatePipelineStateInDb(pipelineID, state);
+    }finally {
```
[the remark this answers is not in the document]
Comment (amaliujia) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/SCMPipelineManager.java:566]: +1

**Comment by amaliujia** [comment]
+1. The lock makes sense.

**Review by adoroszlai — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @Xushaohong for the fix, @amaliujia @bshashikant and @GlenGeng for the review.

**Pull request merged**